src/library_sdl.js

```diff
@@ -1184,6 +1184,10 @@ var LibrarySDL = {
             var v = ({{{ makeGetValue('heapPtr', 'j*numChannels + c', 'i8', 0, 0) }}});
             channelData[j] = ((v >= 0) ? v-128 : v+128) /128;
           }
+        } else if (SDL.audio.format == 0x8120 /*AUDIO_F32*/) {
+          for(var j = 0; j < sizeSamplesPerChannel; ++j) {
+            channelData[j] = ({{{ makeGetValue('heapPtr', '(j*numChannels + c)*4', 'float', 0, 0) }}});
+          }
```

Comment:
perhaps we should add a final `else` that throws an error? not sure why there wans't one before, probably just an oversight.

in theory we could `#ifdef` that assertion-style check by `ASSERTIONS`, but maybe that's overkill.

Reply:
The reason why there are missing else-branches for the supported audio formats might be that this check is already performed in SDL_OpenAudio(), and the format is stored there (so theoretically it cannot be changed in a another place). I have added the missing else+throw in the 2 places and copied the 'invalid format' message from the original throw in SDL_OpenAudio(), the additional check isn't performance-critical in any of the places, and the code is a bit more robust :)